storagenode/pieces/store.go

```diff
@@ -132,20 +136,35 @@ type Store struct {
 	reservedSpace int64
 }
 
+// SpaceUsedLive stores the live totals of used space for all
+// piece content (not including headers) and also total used space
+// by satellites
+type SpaceUsedLive struct {
+	total             int64
+	totalBySatellites map[storj.NodeID]int64
+}
+
 // StoreForTest is a wrapper around Store to be used only in test scenarios. It enables writing
 // pieces with older storage formats and allows use of the ReserveSpace() method.
 type StoreForTest struct {
 	*Store
 }
 
 // NewStore creates a new piece store
-func NewStore(log *zap.Logger, blobs storage.Blobs, v0PieceInfo V0PieceInfoDB, expirationInfo PieceExpirationDB) *Store {
-	return &Store{
+func NewStore(log *zap.Logger, blobs storage.Blobs, v0PieceInfo V0PieceInfoDB, expirationInfo PieceExpirationDB) (*Store, error) {
+	newStore := Store{
 		log:            log,
 		blobs:          blobs,
 		v0PieceInfo:    v0PieceInfo,
 		expirationInfo: expirationInfo,
+		spaceUsedLive:  SpaceUsedLive{},
+	}
+	err := newStore.InitSpaceUsedLive()
```

Comment:
This is not handling context cancellation, since this may take a long time.

Reply:
what migration? this function is iterating over all the piecefiles so i don't think there is a migration for that.

Reply:
added commit to address this comment and have initialization occur after db migration: https://github.com/storj/storj/pull/2707/commits/296ccc47ff0e8cf56f49d541cfe6d56ea595bbda